examples/niscope/graph-measurement.py

```diff
@@ -62,109 +72,119 @@ def CheckStatus(scope_service, vi, result):
+
+    conf_trigger_edge_result = scope_service.ConfigureTriggerEdge(niscope_types.ConfigureTriggerEdgeRequest(
+        vi = vi,
+        trigger_source = channels,
+        level = 0.00,
+        slope = niscope_types.TriggerSlope.TRIGGER_SLOPE_NISCOPE_VAL_POSITIVE,
+        trigger_coupling = niscope_types.TriggerCoupling.TRIGGER_COUPLING_NISCOPE_VAL_DC,
+        holdoff = 0.0
+    ))
+    CheckForError(vi, conf_trigger_edge_result.status)
+
+    result = scope_service.SetAttributeViInt32(niscope_types.SetAttributeViInt32Request(
+        vi = vi,
+        channel_list = channels,
+        attribute_id = niscope_types.NiScopeAttributes.NISCOPE_ATTRIBUTE_MEAS_REF_LEVEL_UNITS,
+        value = niscope_types.NiScopeInt32AttributeValues.NISCOPE_INT32_REF_LEVEL_UNITS_VAL_PERCENTAGE
+    ))
+    CheckForError(vi, result.status)
+
+    # Setup a plot to draw the captured waveform
+    fig = plt.gcf()
+    fig.show()
+    fig.canvas.draw()
+
+    print("\nReading values in loop. CTRL+C to stop.\n")
+    try:
+        while True:
+            # Clear the plot and setup the axis
+            plt.clf()
+            plt.axis([0, 100, -6, 6])
+            # Read a waveform from the scope
+            read_result = scope_service.Read(niscope_types.ReadRequest(
+                vi = vi,
+                channel_list = channels,
+                timeout = 1,
+                num_samples = 10000
+            ))
+            CheckForError(vi, read_result.status)
+            values = read_result.waveform[0:10]
+            print(values)
+
+            # Update the plot with the new waveform
+            plt.plot(read_result.waveform[0:100])
+            fig.canvas.draw()
+            plt.pause(0.001)
+
+            # Fetch the measured average frequency
+            fetch_result = scope_service.FetchMeasurementStats(niscope_types.FetchMeasurementStatsRequest(
+                vi = vi,
+                channel_list = channels,
+                timeout = 1,
+                scalar_meas_function = niscope_types.ScalarMeasurement.SCALAR_MEASUREMENT_NISCOPE_VAL_AVERAGE_FREQUENCY
+            ))
+            CheckForError(vi, fetch_result.status)
+            print("Average Frequency: " + str("%.2f" % round(fetch_result.result[0], 2)) + " Hz")
+            print("")
+
+            time.sleep(0.1)
+    except KeyboardInterrupt:
+        pass
+
+except grpc.RpcError as rpc_error:
+    error_message = rpc_error.details()
+    if rpc_error.code() == grpc.StatusCode.UNAVAILABLE :
+        error_message = f"Failed to connect to server on {server_address}"
+    elif rpc_error.code() == grpc.StatusCode.UNIMPLEMENTED:
+        error_message = "The operation is not implemented or is not supported/enabled in this service"
+    print(f"{error_message}")
+
+finally:
+    if('vi' in vars() and vi.id != 0):
+        # close the session.
+        CheckForError(vi, (scope_service.Close(niscope_types.CloseRequest(
+            vi = vi
+        ))).status)
```

Comment:
In this whole file, try-except block , more exception handling and finally-block is added. Just mentioning here because the diff could be misleading.